core/src/test/scala/org/apache/spark/ContextCleanerSuite.scala

```diff
@@ -206,6 +206,38 @@ class ContextCleanerSuite extends ContextCleanerSuiteBase {
     postGCTester.assertCleanup()
   }
 
+  test("automatically cleanup checkpoint data") {
```

Comment:
Could you also verify that by default (without the configuration set) the checkpoint data is not cleaned?